app/controllers/v2/ManageKeywordsController.scala

```diff
@@ -84,27 +84,77 @@ class ManageKeywordsController @Inject()(
 
   def createKeyword(activeSubNav: SubNavigationTab = ManagerToolsKeywordsTab): Action[AnyContent] =
     (verify.authenticated andThen verify.mustHave(Permission.MANAGE_USERS)).async { implicit request =>
-
-      keywordService.findAll(NoPagination()).flatMap {
-        keywords =>
-          val keywordNames = keywords.results.map(_.name)
-          KeywordForm.formWithAuto(keywordNames).bindFromRequest.fold(
+      keywordService.findAll(NoPagination()).flatMap { keywords =>
+        val keywordNames = keywords.results.map(_.name)
+        KeywordForm
+          .formWithAuto(keywordNames)
+          .bindFromRequest
+          .fold(
             formWithErrors =>
-              Future.successful(BadRequest(
-                newKeywordView(
-                  activeSubNav,
-                  keywords.results,
-                  formWithErrors
-                )
-              )),
+              Future.successful(
+                BadRequest(
+                  newKeywordView(
+                    activeSubNav,
+                    keywords.results,
+                    formWithErrors
+                  )
+                )),
             keyword =>
               keywordService.createKeyword(Keyword(keyword, true)).map { saveKeyword: Keyword =>
                 Redirect(controllers.v2.routes.ManageKeywordsController.displayConfirmKeyword(saveKeyword.name))
-              }
+            }
           )
       }
     }
 
+  def approveOrRejectKeyword(keywordName: String, reference: String): Action[AnyContent] =
+    (verify.authenticated andThen verify.mustHave(Permission.MANAGE_USERS)).async {
+      implicit request: AuthenticatedRequest[AnyContent] =>
+        casesService.getOne(reference).flatMap {
+          case Some(c: Case) => {
+            changeKeywordStatusForm.bindFromRequest.fold(
+              formWithErrors =>
+                Future.successful(
+                  BadRequest(
+                    changeKeywordStatusView(
+                      keywordName,
+                      c,
+                      formWithErrors
+                    )
+                  )),
+              action =>
+                action.toUpperCase match {
+                  case "APPROVE" =>
+                    keywordService.createKeyword(Keyword(keywordName, approved = true)).map {
+                      savedKeyword: Keyword =>
+                         Redirect(
+                           controllers.v2.routes.ManageKeywordsController
+                             .displayKeywordChangeConfirmation(
+                               savedKeyword.name,
+                               savedKeyword.approved,
+                               c.application.goodsName
+                          )
+                      )
+                    }
+                  case "REJECT" =>
```

Comment:
It might be nice to use a model here like `ChangeKeywordStatusAction` or something like that rather than a String but it's not urgent

Reply:
Model created and now in use